ghost/core/core/server/data/db/ConnectionPoolInstrumentation.js

```diff
@@ -0,0 +1,135 @@
+class ConnectionPoolInstrumentation {
+    constructor({knex, logging, metrics, config}) {
+        this.knex = knex;
+        this.pool = this.knex.client.pool;
+        this.logging = logging;
+        this.metrics = metrics;
+        this.enabled = config.get('performanceMonitoring:connectionPool');
+        this.requestStartTimes = {};
+        this.createStartTimes = {};
+    }
+
+    // Get the number of pending creates and acquires plus free and used connections
+    getPoolMetrics() {
+        return {
+            numPendingCreates: this.pool.numPendingCreates(),
+            numPendingAcquires: this.pool.numPendingAcquires(),
+            numFree: this.pool.numFree(),
+            numUsed: this.pool.numUsed()
+        };
+    }
+
+    handleCreateRequest(eventId) {
+        // Track when the request was submitted for calculating wait time
+        this.createStartTimes[eventId] = Date.now();
+        const poolMetrics = this.getPoolMetrics();
+        this.logging.debug(`[ConnectionPool] Creating a connection. EventID: ${eventId} Pending Creates: ${poolMetrics.numPendingCreates} Free: ${poolMetrics.numFree} Used: ${poolMetrics.numUsed}`);
+        this.metrics.metric('connection-pool-create-request', {
```

Comment:
Would it be better to put all of these under a single 'index' (e.g. `metrics-connection-pool` and then add `event: createRequest`)?

I wasn't sure if metrics have to have the same shape within a given 'index' or if they can have different properties.

Reply:
I believe it would be better - one index is easier to deal with in ES than multiple 🙂 AFAIK they don't have to have the same shape

Reply:
Cool cool, updated them all to `metrics-connection-pool`